Engine/ac/overlay.cpp

```diff
@@ -417,6 +425,7 @@ size_t add_screen_overlay_impl(bool roomlayer, int x, int y, int type, int sprnu
     if (type == OVER_CUSTOM) {
         // find an unused custom ID; TODO: find a better approach!
         for (int id = OVER_CUSTOM + 1; (size_t)id <= screenover.size() + OVER_CUSTOM + 1; ++id) {
+            if(id >= overlookup.size()) overlookup.resize(overlookup.size()*2);
```

Comment:
Please disregard what I said earlier in the ticket thread, you having this *almost* in the right place.
But i think you may move it above the loop, and resize to `screenover.size() + OVER_CUSTOM + 2`, this should be the most it needs.

Reply:
pushed a change for this. I kept, but adapted the if clause. It looks like it's working. I still don't know if I need to do something with `recreate_overlay_ddbs`, but I tested loading a save in don't give up the cat as is and it still worked, even after starting the game again.

Reply:
> but I tested loading a save in don't give up the cat as is and it still worked, even after starting the game again.

That is not obligatory, but might speed things up immediately after restoring a save, because the lookup will likely contain outdated indexes, so they all will have to be adjusted dynamically as script accesses the properties.

Reply:
Hey, I gave it a shot [again in a new change](https://github.com/adventuregamestudio/ags/compare/7bc128f4798210884fa3dc1a247ce9cfd132d5ce..8fffaf0b6f844b946dc4bb1ad92c8dd14294fead), it works alright, do you believe it's correct?

Reply:
Yes it looks correct, but I will be able to test only later, after some time.

Also, I said it wrong again, that this is "not obligatory": you must do the "resize" thing after restore anyway, because the save could contain more overlays than before the save. It probably did not make any difference for your game because you are using a fixed size pool of overlays.

Reply:
No rush, when you have time is alright!